packages/native-stack/src/types.tsx

```diff
@@ -545,7 +544,7 @@ export type NativeStackNavigationOptions = {
    *
    * @platform ios
    */
-  sheetAllowedDetents?: SheetDetentTypes;
+  sheetAllowedDetents?: number[] | 'fitToContents';
```

Comment:
This is done just to satisfy typechecker temporality before we merge: https://github.com/react-navigation/react-navigation/pull/12032

If you're not fine with that, let me know.